src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactoryProvider.java

```diff
@@ -89,6 +120,41 @@ public PostgresqlConnectionFactory create(ConnectionFactoryOptions connectionFac
             builder.options(options);
         }
 
+        Boolean ssl = connectionFactoryOptions.getValue(SSL);
+        if (ssl != null) {
+            builder.ssl(ssl);
+        }
+
+        SSLMode sslMode = connectionFactoryOptions.getValue(SSL_MODE);
```

Comment:
Typically, these values come from the URL parser and the URL parser has no knowledge over option types. So expect the value of `SSL_MODE` to potentially be a `String`.